aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -26,11 +26,13 @@ namespace at {
 
 namespace meta {
 
-TORCH_META_FUNC(sin) (
-  const Tensor& self
-) {
-  build_unary_float_op(maybe_get_output(), self);
-}
+#define CREATE_UNARY_META_FUNC(func)                  \
```

Comment:
This is probably fine if most unary ops follow this convention, but any unary ops that, e.g., perform their own error checking shouldn't use this.

Reply:
Would you like me to add a comment?

Reply:
yeah, that couldn't hurt :)

Reply:
You might also have to rebase- I'm seeing merge conflicts in internal tests on `native_functions.yaml`